crates/lint/src/sol/info/unused_import.rs

```diff
@@ -0,0 +1,113 @@
+use solar_ast::{Expr, ExprKind, ImportItems, PathSlice, Symbol, TypeKind, UsingList};
+
+use super::UnusedImport;
+use crate::{
+    declare_forge_lint,
+    linter::{EarlyLintPass, LintContext},
+    sol::{Severity, SolLint},
+};
+
+declare_forge_lint!(
+    UNUSED_IMPORT,
+    Severity::Info,
+    "unused-import",
+    "unused imports should be removed"
+);
+
+impl<'ast> EarlyLintPass<'ast> for UnusedImport {
+    /// Collects all the file imports and caches them in `LintContext`.
+    fn check_import_directive(
+        &mut self,
+        ctx: &mut LintContext<'_>,
```

Comment:
we can avoid &mut context and storing stuff there by adding a `post_source_unit` or similar which runs after walking the source unit